src/transformers/models/whisper/generation_whisper.py

```diff
@@ -798,6 +813,22 @@ def generate_with_fallback(
             for key in ["do_sample", "temperature", "num_beams"]:
                 if key in generate_kwargs:
                     del generate_kwargs[key]
+
+            cur_bsz = decoder_input_ids.shape[0]
+            if generation_config.cache_implementation == "static" and cur_bsz < batch_size:
+                segment_input = F.pad(segment_input, (0, 0, 0, 0, 0, batch_size - cur_bsz), value=0)
+                decoder_input_ids = F.pad(
+                    decoder_input_ids, (0, 0, 0, batch_size - cur_bsz), value=generation_config.pad_token_id
+                )
+                if generate_kwargs.get("decoder_attention_mask") is not None:
+                    generate_kwargs["decoder_attention_mask"] = F.pad(
+                        generate_kwargs["decoder_attention_mask"], (0, 0, 0, batch_size - cur_bsz), value=True
+                    )
+                if generate_kwargs.get("encoder_outputs") is not None:
+                    generate_kwargs["encoder_outputs"] = F.pad(
+                        generate_kwargs["encoder_outputs"], (0, 0, 0, 0, 0, batch_size - cur_bsz), value=0
+                    )
```

Comment:
what are the first 4 0 and then the first 3? (it's counter intuitive to me that batch is last!)

Reply:
```py
F.pad(segment_input, (0, 0, 0, 0, 0, batch_size - cur_bsz), value=0)
```
* First 2 zeroes are for sequence length left/right (no padding)
* Next 2 zeroes are for mel bins left/right (no padding)
* Last zero is for bsz left (no padding)
* The entry `batch_size - cur_bsz` is for bsz right (pad)

It's easy to see with an example:
```python
import torch
import torch.nn.functional as F

# bsz, mel bins, seq len
segment_input = torch.randn(1, 80, 3000)

cur_bsz = segment_input.shape[0]
batch_size = cur_bsz + 1  # demonstrate padding of 1

segment_input = F.pad(segment_input, (0, 0, 0, 0, 0, batch_size - cur_bsz), value=0)
assert segment_input.shape[0] == batch_size
```

Reply:
thanks I checked it out but yeah was counter intutive for me!